pkg/storer/internal/reserve/reserve.go

```diff
@@ -269,12 +276,21 @@ func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 				ChunkType: chunkType,
 				StampHash: stampHash,
 			}),
-			s.ChunkStore().Put(ctx, chunk),
 		)
 		if err != nil {
 			return err
 		}
 
+		if sameAddressSoc {
+			if err := s.ChunkStore().Replace(ctx, chunk); err != nil {
+				return err
+			}
+		} else {
+			if err := s.ChunkStore().Put(ctx, chunk); err != nil {
+				return err
+			}
+		}
```

Comment:
I'll refactor this to get the error from both operations and returns once if there's an error.